Foundation/NSOrderedSet.swift

```diff
@@ -37,38 +48,151 @@ public class NSOrderedSet : NSObject, NSCopying, NSMutableCopying, NSSecureCodin
     
     public required init?(coder aDecoder: NSCoder) { NSUnimplemented() }
     
-    public var count: Int { NSUnimplemented() }
-    public func objectAtIndex(idx: Int) -> AnyObject { NSUnimplemented() }
-    public func indexOfObject(object: AnyObject) -> Int { NSUnimplemented() }
-    public override init() { NSUnimplemented() }
-    public init(objects: UnsafePointer<AnyObject?>, count cnt: Int) { NSUnimplemented() }
+    public var count: Int {
+        return _entries.count
+    }
+
+    public func objectAtIndex(idx: Int) -> AnyObject {
+        return _entryAtIndex(idx).object
```

Comment:
Yes. This was a conscious decision, where I opted to optimize for insertion/deletion time, rather than access-by-index time. With an array, you take a significant hit any time you need to insert or remove an entry from the set.

Do you have a sense of which usage type is better to optimize for?